src/t_stream.c

```diff
@@ -1949,18 +1949,15 @@ void xackCommand(client *c) {
      * error: the return value of this command cannot be an error in case
      * the client successfully acknowledged some messages, so it should be
      * executed in a "all or nothing" fashion. */
+    streamID id[c->argc-3];
```

Comment:
it's very dangerous to create a stack array with size from user input... what if c->argc is 1000000? the stack would overflow

Reply:
@yangbodong22011 same problem: you don't want to allocate huge arrays, even on the heap... in this case the infamous CPU/memory tradeoff should go with the CPU (i.e. leave the code as-is) IMHO

Reply:
@guybe7 `xread` also malloc ids, see [here](https://github.com/redis-io/redis/blob/unstable/src/t_stream.c#L1452).   
Redis limits the length of `multibulk` to `1024 * 1024`, see [here](https://github.com/redis-io/redis/blob/unstable/src/networking.c#L1640), therefore, malloc should not exceed `1024 * 1024 * 16 = 16M`, so it is not a huge memory consumption.

Reply:
@yangbodong22011 indeed you're right
but there's another issue (appears in the comment just above): you don't want `XACK key group 1-0 abc` to ack on 1-0... that's why you return ASAP if one of the IDs is invalid

Reply:
> you don't want `XACK key group 1-0 abc` to ack on 1-0... that's why you return ASAP if one of the IDs is invalid

@guybe7 Sorry, I don't understand what you mean. This PR has no effect on the original logic.

Reply:
@yangbodong22011 my apologies, i guess i'm a bit out of focus today... your PR looks good

Reply:
@guybe7 Thank you for your review.